admin/server/reports.go

```diff
@@ -555,63 +565,46 @@ func (s *Server) generateReportName(ctx context.Context, depl *database.Deployme
 	return uuid.New().String(), nil
 }
 
-func (s *Server) reconcileMagicTokensForExternalEmails(ctx context.Context, orgID, projectID, reportName, ownerID string, emails []string) (map[string]string, error) {
-	emailSet := make(map[string]struct{}, len(emails))
-	for _, email := range emails {
-		emailSet[strings.ToLower(email)] = struct{}{}
-	}
-
-	emailTokens := make(map[string]string, len(emails))
-
-	// check if magic token already exists for the recipient, recipient list might have changed
-	tkns, err := s.admin.DB.FindReportTokensWithSecret(ctx, reportName)
-	if err != nil {
-		if !errors.Is(err, database.ErrNotFound) {
-			return nil, fmt.Errorf("failed to find report tokens: %w", err)
-		}
-	}
-
-	var unusedTknIDs []string
-	for _, tkn := range tkns {
-		if _, ok := emailSet[strings.ToLower(tkn.RecipientEmail)]; ok {
-			id, err := uuid.Parse(tkn.MagicAuthTokenID)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse token ID: %w", err)
-			}
-			token, err := authtoken.FromParts(authtoken.TypeMagic, id, tkn.MagicAuthTokenSecret)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create magic auth token from parts: %w", err)
-			}
-			emailTokens[strings.ToLower(tkn.RecipientEmail)] = token.String()
-			delete(emailSet, tkn.RecipientEmail)
-		} else {
-			unusedTknIDs = append(unusedTknIDs, tkn.MagicAuthTokenID)
-		}
-	}
-
-	// cleanup unused tokens
-	if len(unusedTknIDs) > 0 {
-		err = s.admin.DB.DeleteMagicAuthTokens(ctx, unusedTknIDs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to delete unused report tokens: %w", err)
-		}
-	}
-
-	// return early after cleaning up unused tokens if no external emails
-	if len(emailSet) == 0 {
-		return emailTokens, nil
-	}
-
+func (s *Server) createMagicTokens(ctx context.Context, orgID, projectID, reportName, ownerID, metricsViewName, exploreName, canvasName string, emails []string) (map[string]string, error) {
 	var createdByUserID *string
 	if ownerID != "" {
 		createdByUserID = &ownerID
 	}
+	ttl := 24 * 30 * 6 * time.Hour // approx 6 months
```

Comment:
Let's start with just 3 months

```suggestion
	ttl := 2 * 30 * 24 * time.Hour // approx 3 months
```